crates/fuel-core/src/schema/coins.rs

```diff
@@ -117,7 +117,7 @@ impl MessageCoin {
     async fn asset_id(&self, ctx: &Context<'_>) -> AssetId {
         let params = ctx
             .data_unchecked::<ConsensusProvider>()
-            .latest_consensus_params();
+            .current_consensus_params();
```

Comment:
May I also ask you to replace `Mutex` in the `ConsensusProvider` with `RwLock` please?=)

Since we only read in most of the cases and write happens once per month

Reply:
Done in d8ea573b14fe0092ebd4c6dc75556e223313973e